.github/CONTRIBUTING.md

```diff
@@ -31,141 +31,468 @@ strongly recommended you get approval/traction for it from our forums before sta
+for-loop by specifying a type in the variable of the for's header that DM interprets as a type to filter by. It performs this filter using ```istype()``` (or 
+some internal-magic similar to ```istype()``` - this is BYOND, after all). This is fine in its current state for ```bag_of_items```, but if ```bag_of_items``` 
+contained ONLY swords, or only SUBTYPES of swords, then the above is inefficient. For example:
+```DM
+var/list/bag_of_swords = list(sword, sword, sword, sword)
+var/obj/item/sword/best_sword
+for(var/obj/item/sword/S in bag_of_swords)
+  if(!best_sword || S.damage > best_sword.damage)
+    best_sword = S
+```
+specifies a type for DM to filter by. 
+
+With the previous example that's perfectly fine, we only want swords, but here the bag only contains swords? Is DM still going to try to filter because we gave 
+it a type to filter by? YES, and here comes the inefficiency. Wherever a list (or other container, such as an atom (in which case you're technically accessing 
+their special contents list, but that's irrelevant)) contains datums of the same datatype or subtypes of the datatype you require for your loop's body,
+you can circumvent DM's filtering and automatic ```istype()``` checks by writing the loop as such:
+```DM
+var/list/bag_of_swords = list(sword, sword, sword, sword)
+var/obj/item/sword/best_sword
+for(var/s in bag_of_swords)
+  var/obj/item/sword/S = s
+  if(!best_sword || S.damage > best_sword.damage)
+    best_sword = S
+```
+Of course, if the list contains data of a mixed type then the above optimisation is DANGEROUS, as it will blindly typecast all data in the list as the 
+specified type, even if it isn't really that type, causing runtime errors (AKA your shit won't work if this happens).
+
+#### Dot variable
+Like other languages in the C family, DM has a ```.``` or "Dot" operator, used for accessing variables/members/functions of an object instance.
+eg:
+```DM
+var/mob/living/carbon/human/H = YOU_THE_READER
+H.gib()
+```
+However, DM also has a dot variable, accessed just as `.` on its own, defaulting to a value of null. Now, what's special about the dot operator is that it is automatically returned (as in the `return` statement) at the end of a proc, provided the proc does not already manually return (`return count` for example.) Why is this special?
+
+With `.` being everpresent in every proc, can we use it as a temporary variable? Of course we can! However, the `.` operator cannot replace a typecasted variable - it can hold data any other var in DM can, it just can't be accessed as one, although the `.` operator is compatible with a few operators that look weird but work perfectly fine, such as: `.++` for incrementing `.'s` value, or `.[1]` for accessing the first element of `.`, provided that it's a list.
+
+## Globals versus static
+
+DM has a var keyword, called global. This var keyword is for vars inside of types. For instance:
+
+```DM
+mob
+  var
+    global
+      thing = TRUE
```

Comment:
It is said that we are enforcing absolute pathing. Thus, this should represent that.